code/game/objects/items/storage/fancy.dm

```diff
@@ -159,6 +162,18 @@
 	else
 		to_chat(user, "<span class='notice'>There is nothing left in the pack.</span>")
 
+/obj/item/storage/fancy/cigarettes/CtrlClick(mob/living/carbon/user)
+	if(!istype(user) || !user.canUseTopic(src, BE_CLOSE, ismonkey(user)))
+		return
+	var/obj/item/I = locate(/obj/item/lighter) in contents
+	if(I && contents.len > 0)
+		SEND_SIGNAL(src, COMSIG_TRY_STORAGE_TAKE, I, user)
+		user.put_in_hands(I)
+		contents -= I
+		to_chat(user, "<span class='notice'>You take \a [I] out of the pack.</span>")
+	else
+		to_chat(user, "<span class='notice'>There is no lighter in the pack.</span>")
```

Comment:
```suggestion
		to_chat(user, "<span class='warning'>There is no lighter in the pack.</span>")
```
We're not super consistent about this but IMO successes should be notices and failures should be warnings.